config/config.go

```diff
@@ -60,6 +59,57 @@ type SessionsConfig struct {
 	MaxPerUser uint32
 }
 
+// Counter for the limits  and restrictions for the File Transfer
+//on and off the device(MEN-4325)
+type Counters struct {
+	// Maximum bytes count allowed to transfer per hours
+	// this is per device global limit, which is consulted
+	// every time there is a transfer starting. if above
+	// the limit, we answer with error message indicating
+	// limit reached.
+	MaxBytesTxPerHour uint64
+	MaxBytesRxPerHour uint64
+}
+
+// Limits and restrictions for the File Transfer on and off the device(MEN-4325)
+type FileTransferLimits struct {
+	// the global parent directory that File Transfer will never escape
+	Chroot string
+	// No way to escape Chroot, even if this one is set the Chroot setting will
+	// be checked for the target of any link and restricted accordingly
+	FollowSymLinks bool
+	// Allow overwrite files
+	AllowOverwrite bool
+	// set the owner of new files to OwnerPut
+	OwnerPut string
+	// set the owner of new files to OwnerPut
+	GroupPut string
+	// allow to get only files owned by OwnerGet
+	OwnerGet string
+	// allow to get only files owned by OwnerGet
+	GroupGet string
+	// umask for new files
+	Umask string
+	// Maximum allowed file size
+	MaxFileSize uint64
+	// Global counters
+	Counters Counters
+	// If true it is allowed to upload files with set user id on execute bit set
+	AllowSuid bool
+	// By default we only allow to send/put regular files
+	RegularFilesOnly bool
+	// By default we preserve the file modes but set one according to
+	//the current umask or configured Umask above
+	DoNotPreserveMode bool
+	// By default we preserve the owner of the file uploaded
+	DoNotPreserveOwner bool
+}
+
+type Limits struct {
+	Enabled      bool               `json:"-"`
+	FileTransfer FileTransferLimits `json:"FileTransfer"`
+}
```

Comment:
of course we can. my reasoning:
* there will be other limits in the future and I thought it would be better to have it in the Limits global section, as in: underline the global-security awareness of the configuration
* I kind of thought that "File Transfer Config" is something slightly different than the security limitations and restrictions

The disable limits: will mean that when uninitialised it enforces the limits, is that what we want? (for one I did not want to update every unit test we have)

Reply:
re-reading and thinking about your comment on making the bytes transfer limits global for all transfers to and from mender-connect, perhaps it makes better sense to have Limits as is now? so we can move `MaxBytesTxPerMinute` and `MaxBytesRxPerMinute` one day (see below comments on a follow up task) to Limits and later introduce for instance Limits.PortForward (to limit for instance the range of ports to forward). does it make sense?

Reply:
I won't protest against these design choices, but I would expect (from my "user" perspective) that some of these options (Umask, Owner*, MaxFileSize) to be present under the FileTransfer config object.